cli/commands/queueWithdrawal.go

```diff
@@ -0,0 +1,151 @@
+	Sender      string
+	EstimateGas bool
+	AmountWei   uint64
+}
+
+func QueueWithdrawalCommand(args TQueueWithdrawallArgs) error {
+	ctx := context.Background()
+
+	isSimulation := args.EstimateGas
+
+	eth, err := ethclient.DialContext(ctx, args.EthNode)
+	core.PanicOnError("failed to reach eth node", err)
+
+	chainId, err := eth.ChainID(ctx)
+	core.PanicOnError("failed to load chainId", err)
+
+	acc, err := core.PrepareAccount(&args.Sender, chainId, args.EstimateGas)
+	core.PanicOnError("failed to parse private key", err)
+
+	dm, err := IDelegationManager.NewIDelegationManager(DelegationManager(chainId), eth)
+	core.PanicOnError("failed to reach delegation manager", err)
+
+	pod, err := EigenPod.NewEigenPod(common.HexToAddress(args.EigenPod), eth)
+	core.PanicOnError("failed to reach eigenpod", err)
+
+	_reg, err := pod.WithdrawableRestakedExecutionLayerGwei(nil)
+	core.PanicOnError("failed to load REG", err)
+
+	// [withdrawable]RestakedExecutionlayerWei
+	rew := core.GweiToWei(new(big.Float).SetUint64(_reg))
+	if args.AmountWei > 0 && new(big.Float).SetUint64(args.AmountWei).Cmp(rew) > 0 {
+		return errors.New("invalid --amountWei. must be in the range (0, pod.withdrawableRestakedExecutionLayerGwei() as wei]")
+	}
+
+	podOwner, err := pod.PodOwner(nil)
+	core.PanicOnError("failed to read podOwner", err)
+
+	res, err := dm.GetWithdrawableShares(nil, podOwner, []common.Address{core.BeaconStrategy()})
+	core.PanicOnError("failed to read beacon strategy withdrawable shares", err)
+
+	reg := new(big.Int).SetUint64(_reg)
+
+	sumQueuedWithdrawalsGwei := func() *big.Float {
+		queuedWithdrawals, err := dm.GetQueuedWithdrawals(nil, podOwner)
+		core.PanicOnError("failed to fetch queued withdrawals", err)
+
+		// calculate the 2d-sum of the `shares` array
+		return core.WeiToGwei(utils.BigSum(lo.Map(queuedWithdrawals.Shares, func(values []*big.Int, i int) *big.Int {
+			// return a sum of values.
+			return lo.Reduce(values, func(accum *big.Int, val *big.Int, idx int) *big.Int {
+				return new(big.Int).Add(accum, val)
+			}, big.NewInt(0))
+		})))
+	}()
+
+	// maximumWithdrawalSizeGwei = reg - gwei(sumQueuedWithdrawals)
+	maximumWithdrawalSizeGwei := new(big.Float).Sub(new(big.Float).SetInt(reg), sumQueuedWithdrawalsGwei)
+
+	requestedWithdrawalSizeWei := func() *big.Int {
+		if args.AmountWei == 0 {
+			// if AmountWei isn't specified, we withdraw: all the shares in the beacon strategy
+			return res.WithdrawableShares[0]
+		}
+
+		// if it is specified, we withdraw the specific amount.
+		return new(big.Int).SetUint64(args.AmountWei)
+	}()
+	requestedWithdrawalSizeGwei := core.WeiToGwei(requestedWithdrawalSizeWei)
+
+	if requestedWithdrawalSizeGwei.Cmp(new(big.Float).SetInt(res.WithdrawableShares[0])) > 0 {
+		// requested to withdraw too many shares.
+		return errors.New("the amount to withdraw is larger than the amount of withdrawable shares in the beacon strategy")
+	}
+
+	if requestedWithdrawalSizeGwei.Cmp(maximumWithdrawalSizeGwei) > 0 {
```

Comment:
this check is redundant with the the one above, optionally get rid of comparing with res.WithdrawableShares